### ISSUE TITLE
Missing configuration should be a log warning, not an error

### ISSUE THREAD

**Issue by johnnyshields** [doorkeeper-gem/doorkeeper#1577]
Currently, if you add `gem "doorkeeper"` to your Rails project **without** an initializer, the project will fail to boot with this error:

```
/gems/doorkeeper/lib/doorkeeper/config.rb:33:in `configuration':
Configuration for doorkeeper missing.
Do you have doorkeeper initializer? (Doorkeeper::MissingConfiguration)
```

Instead, this should be simply be a warning log line, which is what all other gems do nearly universally. I've tested on the 200+ gems in my project and Doorkeeper is the only one that causes such a failure.

**Comment by rbclark** [comment]
This is also breaking the `Getting started` steps for doorkeeper rails listed on https://doorkeeper.gitbook.io/guides/ruby-on-rails/getting-started. When running:

```
bundle add doorkeeper
bundle exec rails generate doorkeeper:install
```

The second command gives an error:

```
✗ bundle exec rails generate doorkeeper:install
/Users/rbclark/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/doorkeeper-5.5.4/lib/doorkeeper/config.rb:35:in `configuration': Configuration for doorkeeper missing. Do you have doorkeeper initializer? (Doorkeeper::MissingConfiguration)
```

**Comment by falegk** [comment]
Temporary solution for the issue that @rbclark mentioned
Create a doorkeeper.rb file in the initializers with that content

```
Doorkeeper.configure do
end
```
run `bundle exec rails generate doorkeeper:install` and press Y to overwrite the configuration file.

**Comment by Altonymous** [comment]
Encountered this issue as well.

**Comment by tonybruess** [comment]
+1 this is unexpected behavior

**Comment by CodingItWrong** [comment]
If there is not a plan to fix this soon, I would recommend modifying the Getting Started steps, as they currently don't work due to this issue.

**Comment by nbulaj** [comment]
Hey @johnnyshields  . Could you please try with 

```ruby
# Gemfile

gem "doorkeeper", github: "doorkeeper-gem/doorkeeper", branch: "fixes/lazy-load-associations"
```

And report if issue still present?

**Comment by johnnyshields** [comment]
@nbulaj did you try the sets listed in the issue above on your branch? I don't see why your `lazy-load-associations` PR would fix the behavior of this error message.

**Comment by nbulaj** [comment]
>  I don't see why your lazy-load-associations PR would fix the behavior of this error message. 

Oh don't worry, it's just because you don't understand the root of the issue (which I described somewhere in related threads). But let me repeat - the problem related to the fact that modern Rails or some specific gems (like Sorbet) could load model files **before** initializers have been run. Since some models have constructions like:

```ruby
if Doorkeper.config.blabla
  # do smth
end
```

in their class body - it will trigger an error which you see because `@config` still doesn't exist (`Doorkeeper.configure` wasn't called).

The same is applied to Doorkeeper railtie (engine): it adds some custom parameters filters based on the OAuth configuration, but it does it even for the new Rails apps which have not yet generated an initializer (just added a gem and run `rails g doorkeeper:install`). So the basic steps are:

* Avoid checking config values in railite initializers (config could be nil for Rails apps which just installs the gem)
* Avoid checking config in class definitions 

Maybe one more place missed, but I don't have any of your projects setup so I have no idea what is happening there.

> See comment here: https://github.com/doorkeeper-gem/doorkeeper/pull/1627/files#r1068748108

Sorry, your comment here doesn't help at all. I need to see at least a stack trace to find which line from Doorkeeper internals called something which leaded to the error we're discussing.

> did you try the steps listed in the issue above on your branch?

Could you please which exactly steps you're talking about? Thanks

**Comment by nbulaj** [comment]
@CodingItWrong could you please chech like I suggested there? https://github.com/doorkeeper-gem/doorkeeper/issues/1577#issuecomment-1380925957

**Comment by CodingItWrong** [comment]
@nbulaj I got a chance to give this a try today, and it worked. Following the installation instructions at https://doorkeeper.gitbook.io/guides/ruby-on-rails/getting-started with your branch subbed in, basic testing worked, and I did not run across this error or any other errors :+1:

```
$ ruby -v
ruby 3.2.0 (2022-12-25 revision a528908271) [arm64-darwin22]
$ rails -v
Rails 7.0.4.1
```

**Comment by excid3** [comment]
Just ran into this today. Can confirm that the fixes/lazy-load-associations branch works on Rails 7. 👍

**Comment by nbulaj** [comment]
Should be fixed with 5.6.3 release
